blogService/src/main/java/blogService/service/UserService.java

```diff
@@ -36,12 +72,65 @@ public UserDto getUser(Long userId) {
 
     public String deleteUser(Long userId, String password) {
         Optional<User> optional = userRepository.findById(userId);
-        if (optional.isPresent()) {
-            if (password.hashCode() == userRepository.findById(userId).get().getPassHash()) {
-                userRepository.deleteById(userId);
-                return "The user has been deleted.";
-            }
+        if (optional.isPresent() && password.hashCode() == userRepository.findById(userId).get().getPassHash()) {
+            userRepository.deleteById(userId);
+            return "The user has been deleted.";
         }
         return "The user has not been deleted.";
     }
+
+    public void resetPassword(final String email) {
+        final User user = userRepository.findByEmail(email).orElseThrow(IllegalArgumentException::new);
+        final String temporaryPassword = RandomStringUtils.randomAlphanumeric(6);
+        user.setPassHash(temporaryPassword.hashCode());
+        userRepository.save(user);
+
+        sendMessage(temporaryPassword, email);
+    }
+
+    private void sendMessage(final String newPass, final String email) {
+
+        final Message message = new MimeMessage(setSession());
+        try {
+            message.setFrom(new InternetAddress(mailAddress));
+            message.setRecipients(
+                    Message.RecipientType.TO, InternetAddress.parse(email));
+            message.setSubject("Twoje nowe hasło do konta");
+            String msg = "Twoje hasło do konta zostało zresetowane. Nowe hasło to: " + newPass;
+            MimeBodyPart mimeBodyPart = new MimeBodyPart();
+            mimeBodyPart.setContent(msg, "text/html");
+
+            final Multipart multipart = new MimeMultipart();
+            multipart.addBodyPart(mimeBodyPart);
+
+            message.setContent(multipart);
+
+            Transport.send(message);
+
```

Comment:
odnośnie powyższego komentarz:
tu byłoby `return true;`

Reply:
zrobione

Reply:
skoro zwracasz tę wiadomośc to zwróć ją również w controllerze abyś widziała ją w postmanie :)